### PR TITLE
PRM-2571 Added handler for blank space in TaxReturnStatusEnum

### PR DESCRIPTION
**Pull request by AndrewRButcher**

### CODE REVIEW

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Could you rebase as well please?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Could you rebase as well please?

File: app/viewmodels/IndexPageHelper.scala [hunk 1]
```diff
@@ -55,7 +55,7 @@ class IndexPageHelper @Inject()(p: views.html.components.p,
     val regimeThreshold: Int = penaltiesService.getRegimeThreshold(penaltyDetails)
     val removedPoints: Int = filteredPoints.count(_.expiryReason.nonEmpty)
     val addedPoints: Int = filteredPoints.count(point => point.FAPIndicator.contains("X") && point.penaltyStatus.equals(LSPPenaltyStatusEnum.Active))
-    val amountOfLateSubmissions: Int = filteredPoints.count(_.lateSubmissions.flatMap(_.headOption.map(_.taxReturnStatus.equals(TaxReturnStatusEnum.Open))).isDefined)
+    val amountOfLateSubmissions: Int = filteredPoints.flatMap(point => point.lateSubmissions.map(_.filterNot(lateSubmission => lateSubmission.taxReturnStatus.equals(TaxReturnStatusEnum.AddedFAP)))).flatten.size
```
[the remark this answers is not in the document]
Comment (AndrewRButcher) [reply app/viewmodels/IndexPageHelper.scala:58]: I tried that but it gives this error
[error]  found   : Seq[models.lsp.LateSubmission]
[error]  required: Option[?]
[error]     val amountOfLateSubmissions: Int =filteredPoints.flatMap(point=>point.lateSubmissions.flatMap(_.filterNot(lateSubmission=>lateSubmission.taxReturnStatus.equals(TaxReturnStatusEnum.AddedFAP)))).size

[the remark this answers is not in the document]
Comment (AndrewRButcher) [reply app/viewmodels/IndexPageHelper.scala:58]: There is already a test case for this but I have updated the test data too match what we get from ETMP

[the remark this answers is not in the document]
Comment (LewisMitchell) [reply app/viewmodels/IndexPageHelper.scala:58]: Apologies, my IDE didn't flag it

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**